File: SEP3_UserSystem/src/main/java/model/domain/List/FriendSettingList.java
package model.domain.List;

import model.domain.User.FriendSetting;

import java.util.ArrayList;

public class FriendSettingList {
    private ArrayList<FriendSetting> FriendSettingList;

    public FriendSettingList()
    {
        FriendSettingList = new ArrayList<>();
    }

    public String addNewFriendSetting(FriendSetting newFriend)
    {
        if (getFriendSettingById(newFriend.getId())!=null)
        {
            return "Friend: " + newFriend.getNote() + "[" + newFriend.getId() + "] is not a new friend.";
        }
        else
        {
            FriendSettingList.add(newFriend);
            return null;
        }
    }

    public FriendSetting getFriendSettingById(String id)
    {
        for (int x=0;x<FriendSettingList.size();x++)
        {
            if (FriendSettingList.get(x).getId().equals(id))
            {
                return FriendSettingList.get(x).copy();
            }
        }
        return null;
    }

    public String updateFriendSetting(FriendSetting newFriendSetting)
    {
        for (int x=0;x<FriendSettingList.size();x++)
        {
            if (FriendSettingList.get(x).getId().equals(newFriendSetting.getId()))
            {
                FriendSettingList.get(x).setNote(newFriendSetting.getNote());
                FriendSettingList.get(x).setDisablePost(newFriendSetting.isDisablePost());
                return null;
            }
        }
        return "Can't find the friend: " + newFriendSetting.getNote() + "[" + newFriendSetting.getId() + "]";
    }

    public void removeFriendSettingById(String id)
    {
        for (int x=0;x<FriendSettingList.size();x++)
        {
            if (FriendSettingList.get(x).getId().equals(id))
            {
                FriendSettingList.remove(x);
                break;
            }
        }
    }

    public FriendSettingList copy()
    {
        FriendSettingList copy = new FriendSettingList();
        for (int x=0;x<FriendSettingList.size();x++)
        {
            copy.addNewFriendSetting(FriendSettingList.get(x).copy());
        }
        return copy;
    }
}

File: SEP3_PostSystem/Model/List/UserList/FriendSettingList.cs
using System.Collections.Generic;
using SEP3_PostSystem.Model.Unit.User;

namespace SEP3_PostSystem.Model.List.UserList
{
    public class FriendSettingList
    {
        public List<FriendSetting> FriendSettings { get; set; }

        public FriendSettingList()
        {
            FriendSettings = new List<FriendSetting>();
        }
        
        public string AddNewFriendSetting(FriendSetting newFriend)
        {
            if (newFriend!=null)
            {
                if (GetFriendSettingFromAgreeById(newFriend.GetId())!=null)
                {
                    return "Friend: " + newFriend.GetNote() + "[" + newFriend.GetId() + "] is not a new friend.";
                }
                else
                {
                    FriendSettings.Add(newFriend);
                    return null;
                }
            }

            return "Input null.";
        }

        public FriendSetting GetFriendSettingFromAgreeById(string id)
        {
            FriendSettingList agreeList = GetFriendSettingListByStatus(FriendSettingStatus.AGREE);
            foreach (var friendSetting in agreeList.FriendSettings)
            {
                if (friendSetting.GetId().Equals(id))
                {
                    return friendSetting;
                }
            }
            return null;
        }

        public FriendSetting GetFriendSettingFromUnconfirmedById(string id)
        {
            foreach (var friendSetting in FriendSettings)
            {
                if (friendSetting.NeedAgree()||friendSetting.WaitAgree())
                {
                    if (friendSetting.GetId().Equals(id))
                    {
                        return friendSetting;
                    }
                }
            }
            return null;
        }

        public FriendSetting GetFriendSettingFromNeedAgreeById(string id)
        {
            foreach (var friendSetting in FriendSettings)
            {
                if (friendSetting.NeedAgree()&&friendSetting.GetId().Equals(id))
                {
                    return friendSetting;
                }
            }
            return null;
        }

        public FriendSetting GetFriendSettingFromWaitAgreeById(string id)
        {
            foreach (var friendSetting in FriendSettings)
            {
                if (friendSetting.WaitAgree()&&friendSetting.GetId().Equals(id))
                {
                    return friendSetting;
                }
            }
            return null;
        }
        
        public FriendSettingList GetFriendSettingListById(string id)
        {
            FriendSettingList friendSettingList = new FriendSettingList();
            foreach (var friendSetting in FriendSettings)
            {
                if (friendSetting.GetId().Equals(id))
                {
                    friendSettingList.FriendSettings.Add(friendSetting);
                }
            }
            return friendSettingList;
        }
        
        public FriendSettingList GetFriendSettingListByStatus(FriendSettingStatus status)
        {
            FriendSettingList searchList = new FriendSettingList();
            foreach (var friendSetting in FriendSettings)
            {
                if (friendSetting.GetStatus().Equals(status))
                {
                    searchList.FriendSettings.Add(friendSetting);
                }
            }
            return searchList;
        }
        
        public List<FriendSetting> GetFriendSettingList()
        {
            var copy = new List<FriendSetting>();
            foreach (var friendSetting in FriendSettings)
            {
                copy.Add(friendSetting.Copy());
            }
            return copy;
        }

        public string UpdateFriendSetting(FriendSetting newFriendSetting)
        {
            foreach (var friendSetting in FriendSettings)
            {
                if (friendSetting.GetStatus()==FriendSettingStatus.AGREE)
                {
                    if (friendSetting.GetId().Equals(newFriendSetting.GetId()))
                    {
                        friendSetting.SetNote(newFriendSetting.GetNote());
                        friendSetting.SetDisablePost(newFriendSetting.IsDisablePost());
                        return null;
                    }
                }
            }

            return "Can't find the friend: " + newFriendSetting.GetNote() + "[" + newFriendSetting.GetId() + "]";
        }

        public void RemoveFriendSettingById(string id)
        {
            for (int x = 0; x< FriendSettings.Count; x++)
            {
                if (FriendSettings[x].GetId().Equals(id))
                {
                    FriendSettings.RemoveAt(x);
                    break;
                }
            }
        }

        public FriendSettingList Copy()
        {
            FriendSettingList copy = new FriendSettingList();
            foreach (var friendSetting in FriendSettings)
            {
                copy.FriendSettings.Add(friendSetting.Copy());
            }
            return copy;
        }
    }
}